service/worker/deletenamespace/workflow.go

```diff
@@ -107,6 +107,42 @@ func validateParams(params *DeleteNamespaceWorkflowParams) error {
 	return nil
 }
 
+func validateNamespace(ctx workflow.Context, nsID namespace.ID, nsName namespace.Name, nsClusters []string) error {
+
+	if nsName == primitives.SystemLocalNamespace || nsID == primitives.SystemNamespaceID {
+		return temporal.NewNonRetryableApplicationError("unable to delete system namespace", errors.ValidationErrorErrType, nil, nil)
+	}
+
+	// Disable namespace deletion if namespace is replicate because:
+	//  - If namespace is passive in the current cluster, then WF executions will keep coming from
+	//    the active cluster and namespace will never be deleted (ReclaimResourcesWorkflow will fail).
+	//  - If namespace is active in the current cluster, then it technically can be deleted (and
+	//    in this case it will be deleted from this cluster only because delete operation is not replicated),
+	//    but this is confusing for the users, as they might expect that namespace is deleted from all clusters.
+	if len(nsClusters) > 1 {
+		return temporal.NewNonRetryableApplicationError(fmt.Sprintf("namespace %s is replicated in several clusters [%s]: remove all other cluster and retry", nsName, strings.Join(nsClusters, ",")), errors.ValidationErrorErrType, nil)
+	}
+
+	// NOTE: there is very little chance that another cluster is added after the check above,
+	// but before namespace is marked as deleted below.
+
+	var la *localActivities
+
+	ctx1 := workflow.WithLocalActivityOptions(ctx, localActivityOptions)
+	var bns blockedNamespaces
+	err := workflow.ExecuteLocalActivity(ctx1, la.GetBlockedNamespacesActivity).Get(ctx, &bns)
+	if err != nil {
+		return fmt.Errorf("%w: GetBlockedNamespacesActivity: %v", errors.ErrUnableToExecuteActivity, err)
+	}
+	for _, blockedNsName := range bns.Names {
```

Comment:
use `strings.Contains`?

Reply:
`slices.Contains`, you mean. Yes, forgot about it.